src/safir/testing/kubernetes.py

```diff
@@ -742,6 +824,426 @@ async def list_namespaced_event(
+        name
+            Name of ingress object.
+        namespace
+            Namespace of secret object.
+        body
+            Patches to apply. Only patches with ``op`` of ``replace`` are
+            supported, and only with ``path`` of
+            ``/status/load_balancer/ingress``.
+
+        Returns
+        -------
+        kubernetes_asyncio.client.V1Ingress
+
+        Raises
+        ------
+        kubernetes_asyncio.client.ApiException
+            Raised with 404 status if the ingress does not exist.
+        AssertionError
+            Raised if any other type of patch was provided.
+        """
+        self._maybe_error("patch_namespaced_ingress", name, namespace)
+        ingress = copy.deepcopy(self._get_object(namespace, "Ingress", name))
+        for change in body:
+            assert change["op"] == "replace"
+            assert change["path"] == "/status/load_balancer/ingress"
+            ingress.status.load_balancer.ingress = change["value"]
+        stream = self._event_streams[namespace]["Ingress"]
+        ingress.metadata.resource_version = stream.next_resource_version
+        self._store_object(namespace, "Ingress", name, ingress, replace=True)
+        stream.add_event({"type": "MODIFIED", "object": ingress.to_dict()})
+        return ingress
+
+    # JOB API
+
+    async def create_namespaced_job(self, namespace: str, body: V1Job) -> None:
+        """Create a job object.
+
+        This will in turn create a pod object.  This pod object will have a
+        label "job-name", which will be set to the job's name.  If the mock's
+        ``initial_pod_phase`` is ``Running``, then the Job status will
+        have ``active`` set to 1.
+
+        Parameters
+        ----------
+        namespace
+            Namespace in which to create the object.
+        body
+            Object to create.
+
+        Raises
+        ------
+        kubernetes_asyncio.client.ApiException
+            Raised with 409 status if the object already exists.
+        """
+        self._maybe_error("create_namespaced_job", namespace, body)
+        self._update_metadata(body, "batch/v1", "Job", namespace)
+        name = body.metadata.name
+        self._store_object(namespace, "Job", name, body)
+        # Pretend to spawn a pod
+        # Use our metadata for spawned pod metadata if we weren't given it.
+        template = body.spec.template
+        podmd = template.metadata
+        if podmd is None:
+            podmd = V1ObjectMeta(
+                namespace=namespace,
+                labels=body.metadata.labels.copy(),
+                annotations=body.metadata.annotations.copy(),
+            )
+        podmd.name = name + "-abcde"  # Repeatable for testing
+        podmd.labels["job-name"] = name
+        await self.create_namespaced_pod(
+            namespace, V1Pod(metadata=podmd, spec=template.spec)
+        )
+        pod = await self.read_namespaced_pod(
+            name=podmd.name, namespace=namespace
+        )
```

Comment:
The created pod is modified in place (intentionally), so you can just do:
```suggestion
        pod = V1Pod(metadata=podmd, spec=template.spec)
        await self.create_namespaced_pod(namespace, pod)
```